### PR TITLE
dependency(react-native-paper): added react-native-paper dep, updated babel and added optional require dep for react-native-icons

### PR DESCRIPTION
**Pull request by Calinteodor**

### CODE REVIEW

File: react/features/settings/components/native/SettingsView.js [hunk 1]
```diff
@@ -1,7 +1,8 @@
 // @flow
 
 import React from 'react';
-import { Alert, NativeModules, ScrollView, Switch, Text, TextInput } from 'react-native';
+import { Alert, NativeModules, ScrollView, Text } from 'react-native';
+import { DefaultTheme, Divider, Provider as PaperProvider, Switch, TextInput } from 'react-native-paper';
```
Comment (saghul) [inline react/features/settings/components/native/SettingsView.js:5]: Now that we have base/ui, let's have a JitsiThemeProvider there, like web does.

File: react/features/settings/components/native/SettingsView.js [hunk 2]
```diff
@@ -15,13 +16,26 @@ import {
...
-import FormSectionHeader from './FormSectionHeader';
+import FormSectionAccordion from './FormSectionAccordion';
+import styles, { THUMB_COLOR } from './styles';
 
 /**
  * Application information module.
  */
 const { AppInfo } = NativeModules;
 
+const theme = {
```
Comment (saghul) [inline react/features/settings/components/native/SettingsView.js:27]: Let's have a theme built from the tokens, like web does in base/ui

File: react/features/settings/components/native/styles.js [hunk 3]
```diff
@@ -85,6 +97,23 @@ export default {
...
+        paddingBottom: 8,
+        paddingTop: 2,
+        paddingLeft: 16,
+        paddingRight: 16
+    },
+
+    /**
+     * Text input border style.
+     */
+    textInputBorderColor: ColorPalette.blueHighlight,
```
Comment (saghul) [inline react/features/settings/components/native/styles.js:115]: Shouldn't this be part of the theme?

**Review by saghul — CHANGES_REQUESTED** [review]
Left a couple of comments, PTAL. Mostly, let's start providing stuff in base/ui, like web. Also, screenshot please? :-)

**Comment by saghul** [comment]
Also check the tests please.

**Comment by saghul** [comment]
Jenkins please test this please.

**Comment by Calinteodor** [comment]
Android Expanded
IOS Expanded
Android
IOS

**Review by saghul — DISMISSED** [review]
LGTM! Can you please attach a screenshot?

**Comment by saghul** [comment]
FTR: for some odd reason the bundle size increases the CIs see are not reproducible on macOS. We are investigating.

**Comment by saghul** [comment]
@Calinteodor Upon testing, the text seems cut at the top when typing on the textfields (Android). PTAL.

**Comment by saghul** [comment]
Also, I think the arrows on the accordion are reversed, aren't they? I think it should point downwards when it needs expanding, and upwards when it will collapse.

**Comment by Calinteodor** [comment]
> @Calinteodor Upon testing, the text seems cut at the top when typing on the textfields (Android). PTAL.

**Comment by Calinteodor** [comment]
> > @Calinteodor Upon testing, the text seems cut at the top when typing on the textfields (Android). PTAL.

Can u send me a screenshot please? I tested on master, to see how the text looks inside the old textfields and I didn't see any difference.

**Comment by Calinteodor** [comment]
> Also, I think the arrows on the accordion are reversed, aren't they? I think it should point downwards when it needs expanding, and upwards when it will collapse.

Fixed

**Review by saghul — APPROVED** [review]

**Comment by saghul** [comment]
I was testing with a too large font, nevermind!

**Pull request merged**